face_track/face_track.py

```diff
@@ -376,15 +378,16 @@ def face_event_cb(self, data):
 	def face_loc_cb(self, data):
 		if not self.control_mode & self.C_FACE_TRACKING:
 			return
-		for face in data.faces:
-			fid = face.id
-			loc = face.point
-			# Sanity check.  Sometimes pi_vision sends us faces with
-			# location (0,0,0). Discard these.
-			if loc.x < 0.05:
-				continue
-			self.add_face(fid)
-
+		#below causes error in cmt .. mandeep
```

Comment:
Hi Linas,
 this is a problem because sometimes due to two separate messages of
face_event and face_locations it is sometimes possible that face is removed
via face lost message but it gets re-added due to face_location message.
The callback code however wasn't storing the  face location anywhere, robot
looks by reading a re-broadcast tf2 position of face, which is a improper
hacked way of getting transformed face locations. The better method and
recommended method is to have camera space transform info relative to fixed
point in robot or environment and then calling a transform function to
transform from camera space to world space before storage in map.
Regards,
Mandeep

On Fri, Jul 8, 2016 at 5:42 AM, Linas Vepštas <notifications@github.com>
wrote:

> In face_track/face_track.py
> <https://github.com/opencog/ros-behavior-scripting/pull/53#discussion_r69992979>
> :
>
> > @@ -376,15 +378,16 @@ def face_event_cb(self, data):
> >  	def face_loc_cb(self, data):
> >  		if not self.control_mode & self.C_FACE_TRACKING:
> >  			return
> > -		for face in data.faces:
> > -			fid = face.id
> > -			loc = face.point
> > -			# Sanity check.  Sometimes pi_vision sends us faces with
> > -			# location (0,0,0). Discard these.
> > -			if loc.x < 0.05:
> > -				continue
> > -			self.add_face(fid)
> > -
> > +		#below causes error in cmt .. mandeep
>
> what's the error? why is this a problem?
>
> If this code is stubbed out, then how does the robot know where to look?
>
> —
> You are receiving this because you modified the open/close state.
> Reply to this email directly, view it on GitHub
> <https://github.com/opencog/ros-behavior-scripting/pull/53/files/4d289e1eb160c7a18cc71db993e050a48181c1b7#r69992979>,
> or mute the thread
> <https://github.com/notifications/unsubscribe/AHM_f80sGAu2IPT7QJcC8H_8xcV3puZvks5qTXLJgaJpZM4JEyHb>
> .
>